src/core/generate.test.ts

```diff
@@ -448,8 +516,277 @@ describe("generate", () => {
+    });
+  });
+
+  describe("importing named export types from another file", () => {
+    const sourceText = `
+        import { HeadQuarters } from "./utils";
+        
+        export interface Enemy {
+          name: string;
+          inPrison: boolean;
+          headQuarters: HeadQuarters ;
+        }
+      `;
+    let utilsPath: string;
+    const sourceTextUtils = `
+        export type HeadQuarters = {
+          name: string;
+          location: string;
+        };
+      `;
+    beforeAll(async () => {
+      outputFile = join(tempOpen, "hero");
+      inputPath = join(tempOpen, "hero.ts");
+      utilsPath = join(tempOpen, "utils.ts");
+      await Promise.all([
+        writeFile(inputPath, sourceText),
+        writeFile(utilsPath, sourceTextUtils),
+      ]);
+      const res = await generate({
+        inputPath,
+      });
+
+      getZodSchemasFile = res.getZodSchemasFile;
+      getIntegrationTestFile = res.getIntegrationTestFile;
+      errors = res.errors;
+    });
+
+    it("should generate the zod schemas", () => {
+      expect(getZodSchemasFile(outputFile)).toMatchInlineSnapshot(`
+        "// Generated by ts-to-zod
+        import { z } from \\"zod\\";
+
+        export const headQuartersSchema = z.object({
+            name: z.string(),
+            location: z.string()
+        });
+
+        export const enemySchema = z.object({
+            name: z.string(),
+            inPrison: z.boolean(),
+            headQuarters: headQuartersSchema
+        });
+        "
+      `);
+    });
+
     it("should not have any errors", () => {
       expect(errors).toEqual([]);
     });
   });
+
+  describe("importing aggregated named export types from another file", () => {
+    const sourceText = `
+        import { HeadQuarters } from "./utils";
+        
+        export interface Enemy {
+          name: string;
+          inPrison: boolean;
+          headQuarters: HeadQuarters ;
+        }
+      `;
+    const sourceTextUtils = `
+        type HeadQuarters = {
+          name: string;
+          location: string;
+        };
+        export { HeadQuarters };
+      `;
+    beforeAll(async () => {
+      outputFile = join(tempOpen, "hero");
+      inputPath = join(tempOpen, "hero.ts");
+      const utilsPath = join(tempOpen, "utils.ts");
+      await Promise.all([
+        writeFile(inputPath, sourceText),
+        writeFile(utilsPath, sourceTextUtils),
+      ]);
+      const res = await generate({
+        inputPath,
+      });
+
+      getZodSchemasFile = res.getZodSchemasFile;
+      getIntegrationTestFile = res.getIntegrationTestFile;
+      errors = res.errors;
+    });
+
+    it("should generate the zod schemas", () => {
+      expect(getZodSchemasFile(outputFile)).toMatchInlineSnapshot(`
+        "// Generated by ts-to-zod
+        import { z } from \\"zod\\";
+
+        export const headQuartersSchema = z.object({
+            name: z.string(),
+            location: z.string()
+        });
```

Comment:
I see what you are doing, I tried something else for multiple file support (I was aiming to generate multiple files) but this approach is probably simpler! smart!

Reply:
I also thought about the possibility to generate a multifile structure in a dedicated subdirectory, but this is absolutely easier.